main.tf

```diff
@@ -203,15 +204,15 @@ locals {
 }
 
 resource "aws_eip" "nat" {
-  count = "${var.create_vpc && (var.enable_nat_gateway && !var.reuse_nat_ips) ? (var.single_nat_gateway ? 1 : length(var.azs)) : 0}"
+  count = "${var.create_vpc && (var.enable_nat_gateway && !var.reuse_nat_ips) ? local.nat_gateway_count : 0}"
```

Comment:
This should take into account number of `azs`

Reply:
I think that logic is incorrect. You can pass in a list of AZs, and pass in fewer private subnets. You end up with more NAT Gateways than would be used.

This "fixes" that logic, so you end up with a number of NAT Gateways that matches `max_subnet_length`, or just one if using `single_nat_gateway`.

Happy to move that to another PR if you like though.